tactics/tactics.ml

```diff
@@ -3074,17 +3074,17 @@ let expand_hyp id = Tacticals.New.tclTRY (unfold_body id) <*> clear [id]
 
  *)
 
-let warn_unused_intro_pattern =
+let warn_unused_intro_pattern env =
   CWarnings.create ~name:"unused-intro-pattern" ~category:"tactics"
          (fun names ->
           strbrk"Unused introduction " ++ str (String.plural (List.length names) "pattern")
           ++ str": " ++ prlist_with_sep spc 
          (Miscprint.pr_intro_pattern 
-	 (fun c -> Printer.pr_econstr (snd (c (Global.env()) Evd.empty)))) names)
+	 (fun c -> Printer.pr_econstr (snd (c env Evd.empty)))) names)
 
-let check_unused_names names =
+let check_unused_names names env =
   if not (List.is_empty names) then
-    warn_unused_intro_pattern names
+    warn_unused_intro_pattern env names
```

Comment:
I apparently lost a comment on this line. I was saying that this check should actually take also a sigma, as the following (sorry a bit cabalistic) examples with evars in introduction patterns show:
```coq
Goal exists f, True -> f 0 = 0.
eexists ?[f].
Fail destruct 1 as [H%?f].
```
So, that's good that you passed on this function.